sdk/turing/ensembler.py

```diff
@@ -57,20 +57,27 @@ def initialize(self, artifacts: dict):
 
     def predict(self, context, model_input: pandas.DataFrame) -> \
             Union[numpy.ndarray, pandas.Series, pandas.DataFrame]:
-        prediction_columns = {
-            col: col[len(PyFunc.PREDICTION_COLUMN_PREFIX):]
-            for col in model_input.columns if col.startswith(PyFunc.PREDICTION_COLUMN_PREFIX)
-        }
+        prediction_columns = PyFunc._get_columns_with_header(model_input, PyFunc.PREDICTION_COLUMN_PREFIX)
+        treatment_config_columns = PyFunc._get_columns_with_header(model_input, PyFunc.TREATMENT_CONFIG_COLUMN_PREFIX)
 
         return model_input \
             .rename(columns=prediction_columns) \
+            .rename(columns=treatment_config_columns) \
             .apply(lambda row:
                    self.ensemble(
-                       features=row.drop(prediction_columns.values()),
+                       features=row.drop(prediction_columns.values()).drop(treatment_config_columns.values()),
                        predictions=row[prediction_columns.values()],
-                       treatment_config=None
+                       treatment_config=row[treatment_config_columns.values()]
                    ), axis=1, result_type='expand')
 
+    @staticmethod
+    def _get_columns_with_header(df: pandas.DataFrame, header: str):
```

Comment:
Is it `header` or `prefix`? What's header in this context?

Reply:
Yeah I think `prefix` is the right word I was intending to use; I'll update the names in a bit